src/apps/answers/api.py

```diff
@@ -81,6 +82,12 @@ async def create_answer(
             await AppletHistoryService(session, schema.applet_id, schema.version).get()
         except NotValidAppletHistory:
             raise InvalidVersionError()
+
+        if schema.event_history_id:
+            event = await ScheduleHistoryService(session).get_by_id(schema.event_history_id)
+            if not event:
+                raise NotFoundError("Event not found.")
```

Comment:
I think we could be a bit more descriptive in this error message. Something like
```suggestion
                raise NotFoundError("Invalid event_history_id provided")
```